learning/tour-of-beam/learning-content/java/schema-based-transforms/coder/description.md

```diff
@@ -0,0 +1,120 @@
+<!--
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Coder
+
+A Coder<T> defines how to encode and decode values of type T into byte streams.
+
+Coder instances are serialized during job creation and deserialized before use. This will generally be performed by serializing the object via Java Serialization.
+
+Coder classes for compound types are often composed of coder classes for types contains therein. The composition of Coder instances into a coder for the compound class is the subject of the Coder Provider type, which enables automatic generic composition of Coder classes within the CoderRegistry. See Coder Provider and CoderRegistry for more information about how coders are inferred.
+
+When you write your Custom code, you inherit from Coder and you have to implement its methods.
```

Comment:
When you create custom objects and schemas, you need to create a subclass of `Coder` for your object and implement the following methods: